From: Hector Palacios <hector.palacios@digi.com>
Date: Wed, 14 Jan 2026 16:03:29 +0100
Subject: [PATCH] ccimx95dvk: change SM console from LPUART7 to LPUART2

On version 2 of the SOM, the SM console has changed to LPUART2.

Upstream-Status: Inappropriate [DEY specific]

Signed-off-by: Hector Palacios <hector.palacios@digi.com>

https://onedigi.atlassian.net/browse/DEL-9915
---
 configs/ccimx95dvk.cfg            | 18 +++++++++---------
 configs/ccimx95dvk/config_board.h |  2 +-
 configs/ccimx95dvk/config_scmi.h  | 12 +++++++-----
 configs/ccimx95dvk/config_trdc.h  |  7 +++----
 4 files changed, 20 insertions(+), 19 deletions(-)

diff --git a/configs/ccimx95dvk.cfg b/configs/ccimx95dvk.cfg
index b73401792592..53444e785270 100755
--- a/configs/ccimx95dvk.cfg
+++ b/configs/ccimx95dvk.cfg
@@ -1,7 +1,7 @@
 ## ###################################################################
 ##
 ## Copyright 2023-2025 NXP
-## Copyright 2025 Digi International Inc.
+## Copyright 2025,2026 Digi International Inc.
 ##
 ## Redistribution and use in source and binary forms, with or without modification,
 ## are permitted provided that the following conditions are met:
@@ -42,7 +42,7 @@ include ../devices/MIMX95/configtool/device.cfg
 # Board                                                                    #
 #==========================================================================#
 
-BOARD               DEBUG_UART_INSTANCE=7
+BOARD               DEBUG_UART_INSTANCE=2
 BOARD               DEBUG_UART_BAUDRATE=115200
 
 BOARD               I2C_INSTANCE=1
@@ -233,7 +233,7 @@ IOMUXC              OWNER
 IOMUX_GPR           OWNER
 JTAG                OWNER
 LPI2C1              OWNER
-LPUART7             OWNER
+LPUART2             OWNER
 M33_CACHE_CTRL      OWNER
 M33_PCF             OWNER
 M33_PSF             OWNER
@@ -306,8 +306,8 @@ L_STCU_NPUMIX       OWNER
 PIN_FCCU_ERR0         OWNER
 PIN_I2C1_SCL          OWNER
 PIN_I2C1_SDA          OWNER
-PIN_GPIO_IO08         OWNER  # LPUART7_TX
-PIN_GPIO_IO09         OWNER  # LPUART7_RX
+PIN_UART2_RXD         OWNER
+PIN_UART2_TXD         OWNER
 PIN_WDOG_ANY          OWNER
 
 # Memory
@@ -488,11 +488,11 @@ PERLPI_GPIO3        ALL
 PERLPI_GPIO4        ALL
 PERLPI_GPIO5        ALL
 PERLPI_LPUART1      ALL
-PERLPI_LPUART2      ALL
 PERLPI_LPUART3      ALL
 PERLPI_LPUART4      ALL
 PERLPI_LPUART5      ALL
 PERLPI_LPUART6      ALL
+PERLPI_LPUART7      ALL
 PERLPI_LPUART8      ALL
 PERLPI_GPIO1        ALL
 PERLPI_WDOG3        ALL
@@ -772,11 +772,11 @@ LPTPM4              OWNER
 LPTPM5              OWNER
 LPTPM6              OWNER
 LPUART1             OWNER
-LPUART2             OWNER
 LPUART3             OWNER
 LPUART4             OWNER
 LPUART5             OWNER
 LPUART6             OWNER
+LPUART7             OWNER
 LPUART8             OWNER, test
 LVDS                OWNER
 MIPI_CSI0           OWNER
@@ -922,6 +922,8 @@ PIN_GPIO_IO04        OWNER
 PIN_GPIO_IO05        OWNER
 PIN_GPIO_IO06        OWNER
 PIN_GPIO_IO07        OWNER
+PIN_GPIO_IO08        OWNER
+PIN_GPIO_IO09        OWNER
 PIN_GPIO_IO10        OWNER
 PIN_GPIO_IO11        OWNER
 PIN_GPIO_IO12        OWNER
@@ -987,8 +989,6 @@ PIN_SD3_DATA2        OWNER
 PIN_SD3_DATA3        OWNER
 PIN_UART1_RXD        OWNER, test
 PIN_UART1_TXD        OWNER
-PIN_UART2_RXD        OWNER
-PIN_UART2_TXD        OWNER
 PIN_XSPI1_DATA0      OWNER
 PIN_XSPI1_DATA1      OWNER
 PIN_XSPI1_DATA2      OWNER
diff --git a/configs/ccimx95dvk/config_board.h b/configs/ccimx95dvk/config_board.h
index 23ecae56cf78..b511a0320a68 100644
--- a/configs/ccimx95dvk/config_board.h
+++ b/configs/ccimx95dvk/config_board.h
@@ -54,7 +54,7 @@
 /* Defines */
 
 /*! Config for UART instance */
-#define BOARD_DEBUG_UART_INSTANCE    7U
+#define BOARD_DEBUG_UART_INSTANCE    2U
 /*! Config for UART baudrate */
 #define BOARD_DEBUG_UART_BAUDRATE    115200U
 
diff --git a/configs/ccimx95dvk/config_scmi.h b/configs/ccimx95dvk/config_scmi.h
index 506cb750af65..3b6899a7b9c8 100644
--- a/configs/ccimx95dvk/config_scmi.h
+++ b/configs/ccimx95dvk/config_scmi.h
@@ -177,11 +177,11 @@
         .perlpiPerms[DEV_SM_PERLPI_GPIO4] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_GPIO5] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART1] = SM_SCMI_PERM_ALL, \
-        .perlpiPerms[DEV_SM_PERLPI_LPUART2] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART3] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART4] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART5] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART6] = SM_SCMI_PERM_ALL, \
+        .perlpiPerms[DEV_SM_PERLPI_LPUART7] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART8] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_WDOG3] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_WDOG4] = SM_SCMI_PERM_ALL, \
@@ -270,11 +270,11 @@
         .clkPerms[DEV_SM_CLK_LPSPI7] = SM_SCMI_PERM_ALL, \
         .clkPerms[DEV_SM_CLK_LPSPI8] = SM_SCMI_PERM_ALL, \
         .clkPerms[DEV_SM_CLK_LPUART1] = SM_SCMI_PERM_ALL, \
-        .clkPerms[DEV_SM_CLK_LPUART2] = SM_SCMI_PERM_ALL, \
         .clkPerms[DEV_SM_CLK_LPUART3] = SM_SCMI_PERM_ALL, \
         .clkPerms[DEV_SM_CLK_LPUART4] = SM_SCMI_PERM_ALL, \
         .clkPerms[DEV_SM_CLK_LPUART5] = SM_SCMI_PERM_ALL, \
         .clkPerms[DEV_SM_CLK_LPUART6] = SM_SCMI_PERM_ALL, \
+        .clkPerms[DEV_SM_CLK_LPUART7] = SM_SCMI_PERM_ALL, \
         .clkPerms[DEV_SM_CLK_LPUART8] = SM_SCMI_PERM_ALL, \
         .clkPerms[DEV_SM_CLK_MIPIPHYCFG] = SM_SCMI_PERM_ALL, \
         .clkPerms[DEV_SM_CLK_MIPIPHYPLLBYPASS] = SM_SCMI_PERM_ALL, \
@@ -388,6 +388,8 @@
         .daisyPerms[DEV_SM_DAISY_LPUART6_CTS] = SM_SCMI_PERM_ALL, \
         .daisyPerms[DEV_SM_DAISY_LPUART6_RXD] = SM_SCMI_PERM_ALL, \
         .daisyPerms[DEV_SM_DAISY_LPUART6_TXD] = SM_SCMI_PERM_ALL, \
+        .daisyPerms[DEV_SM_DAISY_LPUART7_RXD] = SM_SCMI_PERM_ALL, \
+        .daisyPerms[DEV_SM_DAISY_LPUART7_TXD] = SM_SCMI_PERM_ALL, \
         .daisyPerms[DEV_SM_DAISY_NETC_EMDC] = SM_SCMI_PERM_ALL, \
         .daisyPerms[DEV_SM_DAISY_NETC_EMDIO] = SM_SCMI_PERM_ALL, \
         .daisyPerms[DEV_SM_DAISY_NETC_ETH0_RMII_RX_ER] = SM_SCMI_PERM_ALL, \
@@ -463,11 +465,11 @@
         .perlpiPerms[DEV_SM_PERLPI_GPIO4] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_GPIO5] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART1] = SM_SCMI_PERM_ALL, \
-        .perlpiPerms[DEV_SM_PERLPI_LPUART2] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART3] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART4] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART5] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART6] = SM_SCMI_PERM_ALL, \
+        .perlpiPerms[DEV_SM_PERLPI_LPUART7] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_LPUART8] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_WDOG3] = SM_SCMI_PERM_ALL, \
         .perlpiPerms[DEV_SM_PERLPI_WDOG4] = SM_SCMI_PERM_ALL, \
@@ -515,6 +517,8 @@
         .pinPerms[DEV_SM_PIN_GPIO_IO05] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_GPIO_IO06] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_GPIO_IO07] = SM_SCMI_PERM_ALL, \
+        .pinPerms[DEV_SM_PIN_GPIO_IO08] = SM_SCMI_PERM_ALL, \
+        .pinPerms[DEV_SM_PIN_GPIO_IO09] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_GPIO_IO10] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_GPIO_IO11] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_GPIO_IO12] = SM_SCMI_PERM_ALL, \
@@ -580,8 +584,6 @@
         .pinPerms[DEV_SM_PIN_SD3_DATA3] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_UART1_RXD] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_UART1_TXD] = SM_SCMI_PERM_ALL, \
-        .pinPerms[DEV_SM_PIN_UART2_RXD] = SM_SCMI_PERM_ALL, \
-        .pinPerms[DEV_SM_PIN_UART2_TXD] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_XSPI1_DATA0] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_XSPI1_DATA1] = SM_SCMI_PERM_ALL, \
         .pinPerms[DEV_SM_PIN_XSPI1_DATA2] = SM_SCMI_PERM_ALL, \
diff --git a/configs/ccimx95dvk/config_trdc.h b/configs/ccimx95dvk/config_trdc.h
index 12ebbc61ca64..5758f62bdef5 100644
--- a/configs/ccimx95dvk/config_trdc.h
+++ b/configs/ccimx95dvk/config_trdc.h
@@ -108,7 +108,7 @@
         SM_CFG_W1(0x00010450U), 0x30003330U, \
         SM_CFG_W1(0x00010454U), 0x03333033U, \
         SM_CFG_W1(0x00010458U), 0x00030000U, \
-        SM_CFG_W1(0x0001045cU), 0x33330000U, \
+        SM_CFG_W1(0x0001045cU), 0x33330030U, \
         SM_CFG_W1(0x00010460U), 0x33333333U, \
         SM_CFG_W1(0x00010464U), 0x30000003U, \
         SM_CFG_W1(0x00010468U), 0x33330030U, \
@@ -123,7 +123,7 @@
         SM_CFG_W1(0x00010650U), 0x09000309U, \
         SM_CFG_W1(0x00010654U), 0x0000C900U, \
         SM_CFG_W1(0x00010658U), 0x99909900U, \
-        SM_CFG_W1(0x0001065cU), 0x00009999U, \
+        SM_CFG_W1(0x0001065cU), 0x00009909U, \
         SM_CFG_W1(0x00010668U), 0x00009900U, \
         SM_CFG_W1(0x0001066cU), 0x90909000U, \
         SM_CFG_W1(0x00010670U), 0x00009000U, \
@@ -1151,7 +1151,6 @@
         SM_CFG_Z1(0x00010260U), \
         SM_CFG_W1(0x00010270U), 0x00000090U, \
         SM_CFG_W1(0x00010460U), 0x33000300U, \
-        SM_CFG_W1(0x00010474U), 0x00000030U, \
         SM_CFG_W1(0x00010478U), 0x33300000U, \
         SM_CFG_W1(0x0001047cU), 0x00003333U, \
         SM_CFG_W1(0x000105d0U), 0x33333333U, \
@@ -1170,7 +1169,7 @@
         SM_CFG_W1(0x00010668U), 0x99999999U, \
         SM_CFG_W1(0x0001066cU), 0x09999999U, \
         SM_CFG_W1(0x00010670U), 0x99900009U, \
-        SM_CFG_W1(0x00010674U), 0x99999909U, \
+        SM_CFG_W1(0x00010674U), 0x99999999U, \
         SM_CFG_W1(0x00010678U), 0x00009999U, \
         SM_CFG_W1(0x0001067cU), 0x00990000U, \
         SM_CFG_W1(0x00010780U), 0x00000009U, \
